lib/matplotlib/backends/backend_ps.py

```diff
@@ -1150,13 +1150,8 @@ def convert_psfrags(tmpfile, psfrags, font_preamble, custom_preamble,
     with mpl.rc_context({
             "text.latex.preamble":
             mpl.rcParams["text.latex.preamble"] +
-            # Only load these packages if they have not already been loaded, in
-            # order not to clash with custom packages.
-            r"\makeatletter"
-            r"\@ifpackageloaded{color}{}{\usepackage{color}}"
-            r"\@ifpackageloaded{graphicx}{}{\usepackage{graphicx}}"
-            r"\@ifpackageloaded{psfrag}{}{\usepackage{psfrag}}"
-            r"\makeatother"
+            "".join(map(mpl.texmanager._usepackage_if_not_loaded,
```

Comment:
I'd still write this out as three explicit function calls. That's one line longer but sill easier to read follow than join/map.

Reply:
sure